File: study/ts/script/data/UIGuideData.ts
import { BaseData } from "./BaseData";
import { G_UserData, G_ConfigLoader } from "../init";
import { ConfigNameConst } from "../const/ConfigNameConst";
import { UIGuideUnitData } from "./UIGuideUnitData";

export interface UIGuideData {
}
let schema = {};
export class UIGuideData extends BaseData {
    public static schema = schema;

    constructor(properties?) {
        super(properties);
    }
    public clear() {
    }
    public reset() {
    }
    public hasUIGuide(guideType, paramValue) {
        let cfg = this.findCfg(guideType, paramValue);
        if (!cfg) {
            return false;
        }
        return true;
    }
    public needShowGuide(guideType, paramValue) {
        let cfg = this.findCfg(guideType, paramValue);
        if (!cfg) {
            return false;
        }
        let level = G_UserData.getBase().getLevel();
        if (level < cfg.level_min || level > cfg.level_max) {
            return false;
        }
        return true;
    }
    public findCfg(guideType, paramValue) {
        paramValue = paramValue || 0;
        let level = G_UserData.getBase().getLevel();
        let guideCfg = null;
        let StoryStageGuide = G_ConfigLoader.getConfig(ConfigNameConst.STORY_STAGE_GUIDE);
        for (let i = 0; i < StoryStageGuide.length(); i += 1) {
            let cfg = StoryStageGuide.indexOf(i);
            if (cfg.type == guideType && cfg.value == paramValue) {
                guideCfg = cfg;
                break;
            }
        }
        return guideCfg;
    }
    public createUIGuideUnitData(guideType, paramValue) {
        let guideCfg = this.findCfg(guideType, paramValue);
        let unitData = new UIGuideUnitData();
        unitData.initData(guideCfg);
        return unitData;
    }
}

File: study/lua/src/app/data/UIGuideData.lua
local BaseData = require("app.data.BaseData")
local UIGuideData = class("UIGuideData", BaseData)
local StoryStageGuide = require("app.config.story_stage_guide")
local UIGuideUnitData = import(".UIGuideUnitData")
local schema = {}
UIGuideData.schema = schema
function UIGuideData:ctor(properties)
    UIGuideData.super.ctor(self, properties)
end

--清除
function UIGuideData:clear()
end

function UIGuideData:reset()
end

function UIGuideData:hasUIGuide(guideType,paramValue)
    local cfg = self:findCfg(guideType,paramValue)
    if not cfg then
        return false
    end
    return true
end

function UIGuideData:needShowGuide(guideType,paramValue)
    local cfg = self:findCfg(guideType,paramValue)
    if not cfg then
        return false
    end

    local level = G_UserData:getBase():getLevel() 
    if level < cfg.level_min or level > cfg.level_max then
        return false
    end
    
    return true
end

function UIGuideData:findCfg(guideType,paramValue)
    paramValue = paramValue or 0
    local level = G_UserData:getBase():getLevel() 
    local guideCfg = nil
    for i = 1,StoryStageGuide.length(),1 do
        local cfg = StoryStageGuide.indexOf(i)
        if cfg.type == guideType and  cfg.value == paramValue then
            guideCfg =  cfg
            break
        end
    end
   return guideCfg
end

function UIGuideData:createUIGuideUnitData(guideType,paramValue)
    local guideCfg = self:findCfg(guideType,paramValue)
    local unitData = UIGuideUnitData.new()
    unitData:initData(guideCfg)
    return unitData
end


return UIGuideData